core/src/main/java/org/apache/calcite/rel/rules/FilterMultiJoinMergeRule.java

```diff
@@ -43,7 +44,9 @@
   //~ Constructors -----------------------------------------------------------
 
   /**
-   * Creates a FilterMultiJoinMergeRule.
+   * Creates a FilterMultiJoinMergeRule that uses {@link Filter}
+   * of type {@link LogicalFilter}
+   * @param relBuilderFactory builder factory for relational expressions
    */
   public FilterMultiJoinMergeRule(RelBuilderFactory relBuilderFactory) {
     super(
```

Comment:
I think it is better to use the telescoping constructor pattern here.

Reply:
done